packages/SwingSet/test/test-liveslots.js

```diff
@@ -461,12 +458,12 @@ async function doResultPromise(t, mode) {
   } else {
     throw Error(`unknown mode ${mode}`);
   }
-  await endOfCrank();
+  await waitUntilQuiescent();
   t.deepEqual(log, []);
 
   // Now we resolve p2, allowing the second two() to proceed
   dispatch.notifyFulfillToData(expectedP2, capargs(4, []));
-  await endOfCrank();
+  await waitUntilQuiescent();
```

Comment:
Is quiescent different than idle?

Reply:
Hm. In this case we use "quiescent" to mean there is nothing left on the Promise queue, so the vat (which is isolated from everything except the syscall/dispatch pathway) has lost all agency, and cannot obtain control again until the next dispatch invocation. There may still be things on the IO/timer queue, inbound events and such, but only the kernel and/or devices can observe those, not regular vats.

"Idle" might imply that the entire process has nothing to do.

We use the timer queue (and the fact that it is serviced with lower priority than the Promise queue) to implement `waitUntilQuiescent` (with a `setImmediate`). Vats are not given access to timers or `setImmediate`, or any other IO.